src/core/tests/Test_CoreMain.py

```diff
@@ -1143,6 +1168,624 @@ def __check_telemetry_events(self, runtime):
+        self.assertEqual(len(json.loads(substatus_file_data["formattedMessage"]["message"])["errors"]["details"]), 0)
+        self.assertFalse("review this log file on the machine" in json.loads(substatus_file_data["formattedMessage"]["message"])["errors"]["message"])
+        # self.assertEqual(len(runtime.status_handler._StatusHandler__assessment_packages_removed), 0)
+
+        runtime.stop()
+
+    def test_assessment_operation_truncation_over_size_limit(self):
+        argument_composer = ArgumentComposer()
+        argument_composer.operation = Constants.ASSESSMENT
+        runtime = RuntimeCompositor(argument_composer.get_composed_arguments(), True, Constants.ZYPPER)
+        runtime.set_legacy_test_type('HappyPath')
+        CoreMain(argument_composer.get_composed_arguments())
+
+        # Check telemetry events
+        self.__check_telemetry_events(runtime)
+
+        # HappyPath add 3 additional packages
+        # {\"patchId\": \"kernel-default_4.4.49-92.11.1_Ubuntu_16.04\", \"name\": \"kernel-default\", \"version\": \"4.4.49-92.11.1\", \"classifications\": [\"Security\"]},
+        # {\"patchId\": \"libgcc_5.60.7-8.1_Ubuntu_16.04\", \"name\": \"libgcc\", \"version\": \"5.60.7-8.1\", \"classifications\": [\"Other\"]},
+        # {\"patchId\": \"libgoa-1_0-0_3.20.5-9.6_Ubuntu_16.04\", \"name\": \"libgoa-1_0-0\", \"version\": \"3.20.5-9.6\", \"classifications\": [\"Other\"]}
+
+        patch_count = random.randint(780, 1000)
+        test_packages, test_package_versions = self.__set_up_packages_func(patch_count)
+        runtime.status_handler.set_package_assessment_status(test_packages, test_package_versions, "Critical")
+        runtime.status_handler.set_assessment_substatus_json(status=Constants.STATUS_SUCCESS)
+
+        # Assert complete status file
+        with runtime.env_layer.file_system.open(runtime.execution_config.complete_status_file_path, 'r') as file_handle:
+            substatus_file_data = json.load(file_handle)
+
+        self.assertTrue(len(json.dumps(substatus_file_data)) > Constants.StatusTruncationConfig.AGENT_FACING_STATUS_FILE_SIZE_LIMIT_IN_BYTES)
+        self.__assert_complete_status_output(substatus_file_data, Constants.ASSESSMENT, Constants.PATCH_ASSESSMENT_SUMMARY, Constants.STATUS_SUCCESS, patch_count + 3)
+
+        # Assert assessment truncated status file
+        with runtime.env_layer.file_system.open(runtime.execution_config.status_file_path, 'r') as file_handle:
+            substatus_file_data = json.load(file_handle)
+
+        self.assertTrue(len(json.dumps(substatus_file_data)) < Constants.StatusTruncationConfig.AGENT_FACING_STATUS_FILE_SIZE_LIMIT_IN_BYTES)
+        self.__assert_truncated_status_output(substatus_file_data[0]["status"]["substatus"][0], Constants.PATCH_ASSESSMENT_SUMMARY, Constants.STATUS_WARNING)
+        # Assert assessment tombstone
+        self.__assert_truncated_assessment_tombstone(substatus_file_data[0]["status"]["substatus"][0], patch_count + 3)
```

Comment:
We don't have any changes regarding tombstones in this PR. Why the assertion then?

Reply:
because in this function there's assertion checking if the new message patches < original, this indicates truncation did occurred and the message patches param will be use checking if there's tombstone record as well.

open for suggestions

Reply:
Code does not exist now